IPA-End-to-End.md

```diff
@@ -1924,6 +1927,80 @@ An example of an oblivious algorithm is the bitonic sorting algorithm [[1]](http
+   Examples of these include [Windows WebView2](https://learn.microsoft.com/en-us/windows/apps/design/controls/web-view),
+   [iOS WKWebView](https://developer.apple.com/documentation/webkit/wkwebview), or
+   [Android WebView](https://developer.android.com/develop/ui/views/layout/webapps).
+
+2. Where the operation of the browser component is not accessible to the application.
+   Examples of this approach are [Android Custom Tabs](https://developer.chrome.com/docs/android/custom-tabs/) and
+   [SFSafariViewController](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller).
+
+In all cases, IPA provides the greatest benefit if there is a single system-wide (or user-wide) store of match keys.
+To facilitate this, support for storing and managing match keys at the level of the operating system provides the greatest utility.
+A system-wide utility will ensure that attribution can work across different apps, including browsing.
+Without a system-wide store, attribution can still work, though it will be limited to impressions and conversions in the same app.
+
+
+### Applications that can access web content
+
+If the embedded browser makes content accessible to the application, then the application is effectively acting as an independent browser.
+An application therefore needs distinct storage from the regular browser so that one application cannot access private information hosted by another application or browser.
+
+When IPA integrates into this sort of environment, the `getEncryptedMatchKey()` API needs to return an encryption of the match key that is specific to the combination of application and site.
+The same value cannot be used across different apps or sites, or the ciphertext that is provided might link activity across contexts.
+Integration of the IPA API at the system level therefore needs to segment any storage of _encrypted match keys_ based on the identity of the application that requests it, in addition to the site.
+
+Note that this is an imperfect model because activity on sites shown in a web view might be linkable to activity in browser or other apps.
+Many sites attempt to detect webviews and disable login, which can prevent more direct forms of linkability.
+However, some applications use embedded browsers to manage login and authorization of access to online services.
+Consequently, people can become accustomed to having to log in, despite that being a poor security practice, and in doing so create strong correlation between contexts.
+
+Aside from partitioning the storage of _encrypted match keys_ by app, no other adjustment is being considered.
+Sites can use match keys as usual.
+The app will be able to view or even change the encrypted match key, but the value will not reveal anything.
+This access might allow actions that could be seen to be malicious, such as:
+
+* An app can violate user privacy by _encrypted match keys_ to different sites that allow those sites to link user activity.
```

Comment:
Something is amiss here. I've read it 4 times and I don't understand. Missing a few words perhaps?